### ISSUE TITLE
IDE0001 (Simplify type name) suggests removing Nullable<T> where behavior changes

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#72420]
**Version Used**: 

**Steps to Reproduce**:

![image](https://github.com/dotnet/roslyn/assets/1408396/0781dc94-9387-4aed-87f7-40a28b3355e5)

```cs
public void T()
{
    static void test<T>(Func<T> func)
    {
    }

    // IDE0001 expected for this call:
    test<UnmanagedCustomStruct>(() => new UnmanagedCustomStruct());

    // IDE0001 NOT expected for this call
    test<UnmanagedCustomStruct?>(() => new UnmanagedCustomStruct());
}

public struct UnmanagedCustomStruct
{
    public Guid Foo;
    public int Bar;
}
```

**Diagnostic Id**:

If this is a report about a bug in an analyzer, please include the diagnostic if possible (e.g. `"IDE0030"`).

**Expected Behavior**:

No diagnostic reported.
Required generic argument is not removed.

**Actual Behavior**:

IDE0001 is reported, and suggests removing a generic type argument that causes a behavior change.

**Comment by CyrusNajmabadi** [comment]
Speculation analyzer should definitely be validating that the final instantiated call has the same type args as the original.  That's bad if it doesn't :)
